pkg/vpcmodel/drawioOutput.go

```diff
@@ -59,62 +61,86 @@ func (d *DrawioOutputFormatter) createDrawioTree() {
 
 func (d *DrawioOutputFormatter) createNodeSets() {
 	for _, vpcConfig := range d.cConfigs {
-		for _, ns := range vpcConfig.NodeSets {
-			if d.showResource(ns) {
-				d.gen.TreeNode(ns)
+		if !vpcConfig.IsMultipleVPCsConfig {
+			for _, ns := range vpcConfig.NodeSets {
+				if d.showResource(ns) {
+					d.gen.TreeNode(ns)
+				}
 			}
 		}
 	}
 }
 
 func (d *DrawioOutputFormatter) createNodes() {
 	for _, vpcConfig := range d.cConfigs {
-		for _, n := range vpcConfig.Nodes {
-			if !n.IsExternal() && d.showResource(n) {
-				d.gen.TreeNode(n)
+		if !vpcConfig.IsMultipleVPCsConfig {
+			for _, n := range vpcConfig.Nodes {
+				if !n.IsExternal() && d.showResource(n) {
+					d.gen.TreeNode(n)
+				}
 			}
 		}
 	}
 }
 
 func (d *DrawioOutputFormatter) createFilters() {
 	for _, vpcConfig := range d.cConfigs {
-		for _, fl := range vpcConfig.FilterResources {
-			if d.showResource(fl) {
-				d.gen.TreeNode(fl)
+		if !vpcConfig.IsMultipleVPCsConfig {
+			for _, fl := range vpcConfig.FilterResources {
+				if d.showResource(fl) {
+					d.gen.TreeNode(fl)
+				}
 			}
 		}
 	}
 }
 
 func (d *DrawioOutputFormatter) createRouters() {
-	for _, vpcConfig := range d.cConfigs {
+	for vpcResourceName, vpcConfig := range d.cConfigs {
 		for _, r := range vpcConfig.RoutingResources {
 			if d.showResource(r) {
 				rTn := d.gen.TreeNode(r)
-				for _, ni := range r.Src() {
-					if d.showResource(ni) {
-						d.routers[d.gen.TreeNode(ni)] = rTn.(drawio.IconTreeNodeInterface)
+				if vpcConfig.IsMultipleVPCsConfig {
+					d.multiVpcRouters[vpcResourceName] = rTn.(drawio.IconTreeNodeInterface)
+				} else {
+					for _, ni := range r.Src() {
+						if d.showResource(ni) {
+							d.nodeRouters[d.gen.TreeNode(ni)] = rTn.(drawio.IconTreeNodeInterface)
```

Comment:
`d.routers` was renamed to `d.nodeRouters`. Right?

Reply:
yes